argoverse/evaluation/detection_utils.py

```diff
@@ -0,0 +1,135 @@
+from enum import Enum, auto
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from scipy.spatial.transform import Rotation as R
+
+from argoverse.data_loading.object_label_record import ObjectLabelRecord
+from argoverse.utils.transform import quat2rotmat
+
+
+class SimFnType(Enum):
+    CENTER = auto()
+    IOU_2D = auto()
+    IOU_3D = auto()
+
+
+class DistFnType(Enum):
+    TRANSLATION = auto()
+    SCALE = auto()
+    ORIENTATION = auto()
+
+
+def filter_annos(annos: np.ndarray, target_class: str, range_metric="euclidean", max_dist=50) -> np.ndarray:
+    """Filter the annotations based on a set of conditions.
+
+    Args:
+        annos: The annotations to be filtered.
+        target_class: The name of the class of interest.
+        range_metric: The range metric used for filtering.
+    
+    Returns:
+        The filtered annotations.
+    """
+    annos = np.array([dt_anno for dt_anno in annos if dt_anno.label_class == target_class])
```

Comment:
Happy to change this, but can we enforce this as a pre-commit hook?

Reply:
Yeah so I went and fixed all the line lengths to <= 80, and then the existing pre-commit hooks reverted all those changes. Not sure how to fix that

Reply:
Looks like 120 chars is our prec-ommit max length. Matt and Guillaume didn't think it made sense to shorten that  because it would create a lot of useless churn in the repo. Looks like the lines were wrapping around strangely for me just because I was looking at a small laptop screen. I think 120 chars is fine after all.